### PR TITLE
hwmv2: Port STM32F2X SoCs & board

### PR DESCRIPTION
**Pull request by FRASTM**
Port STM32F2X SoCs & board to Hardware Model V2.

- nucleo_f207zg

### CODE REVIEW

**Comment by FRASTM** [comment]
rebase on remotes/origin/collab-hwm 1966f38ea2cebfd58261a5197f5d188a8794accd

File: soc/st/stm32/stm32f2/Kconfig [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 
 # Copyright (c) 2018 qianfan Zhao <qianfanguijin@163.com>
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_SERIES_STM32F2X
-	bool "STM32F2x Series MCU"
 	select ARM
 	select CPU_CORTEX_M3
 	select CPU_CORTEX_M_HAS_DWT
 	select SOC_FAMILY_STM32
```
Comment (erwango) [inline soc/st/stm32/stm32f2/Kconfig:10]: Remove `select SOC_FAMILY_STM32`

**Review by erwango — CHANGES_REQUESTED** [review]
Please also fix path in .rst file: `boards/arm/nucleo_f207zg/nucleo_f207zg_defconfig`

File: soc/st/stm32/stm32f2/Kconfig (hunk 1 above)
Comment (FRASTM) [reply soc/st/stm32/stm32f2/Kconfig]: done

File: boards/st/nucleo_f207zg/doc/index.rst [hunk 2]
```diff
@@ -112,7 +112,7 @@ The Zephyr nucleo_207zg board configuration supports the following hardware feat
 Other hardware features are not yet supported on this Zephyr port.
 
 The default configuration can be found in the defconfig file:
-``boards/arm/nucleo_f207zg/nucleo_f207zg_defconfig``
+``boards/st/nucleo_f207zg/nucleo_f207zg_defconfig``
```
Comment (ajarmouni-st) [inline boards/st/nucleo_f207zg/doc/index.rst:115]:
```suggestion
- :zephyr_file:`boards/st/nucleo_f207zg/nucleo_f207zg_defconfig`
```

Comment (FRASTM) [reply boards/st/nucleo_f207zg/doc/index.rst]: done

**Review by erwango — APPROVED** [review]

File: soc/st/stm32/stm32f2/Kconfig.soc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_STM32F2X
+	bool
+	select SOC_FAMILY_STM32
+
+config SOC_SERIES
+	default "stm32f2" if SOC_SERIES_STM32F2X
+
+config SOC_SERIES_STM32F205XX
```
Comment (erwango) [inline soc/st/stm32/stm32f2/Kconfig.soc:13]:
```suggestion
config SOC_STM32F205XX
```

Comment (erwango) [reply soc/st/stm32/stm32f2/Kconfig.soc:13]: Should fix CI

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by tejlmand — APPROVED** [review]
Please address comment by @erwango.

Rest looks fine, so approving as @erwango has already pointed out the root cause of CI failure.

File: soc/st/stm32/stm32f2/Kconfig (hunk 1 above)
Comment (erwango) [reply soc/st/stm32/stm32f2/Kconfig:10]: Not done. This is the cause of the dependency loop reported in CI.

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by ajarmouni-st — APPROVED** [review]

**Pull request merged**